docs/book/beyond_basics/sample_webhook.md

```diff
@@ -0,0 +1,207 @@
+# Webhook Example
+
+This chapter walks through a simple webhook implementation.
+
+It uses the [controller-runtime](https://godoc.org/sigs.k8s.io/controller-runtime/pkg) libraries to implement
+the Webhook and Manager. The webhook can run with other controllers in the same manager.
+
+## Setup
+
+#### Way to Deploy your Webhook Server
+
+There are various ways to deploy the webhook server in terms of
+
+1. Where the serving certificates live.
+1. In what environment the webhook server runs, in a pod or directly on a VM, etc.
+1. If in a pod, on what type of node, worker nodes or master node.
+
+The recommended way to deploy the webhook server is
+
+1. Run the webhook server as a regular pod on worker nodes through a workload API, e.g. Deployment or StatefulSet.
+1. Put the certificate in a k8s secret in the same namespace as the webhook server
+1. Mount the secret as a volume in the pod
+1. Create a k8s service to front the webhook server.
+
+#### Creating a Webhook Handler
+
```

Comment:
Introduce the Webhook Handler.

`
The business logic for a Webhook written by implementing the `admission.Handler` interface.  This interface contains a single `Handle` method that is invoked for each request the Webhook is configured to accept.
`